chemprop/models/model.py

```diff
@@ -1,160 +1,341 @@
-from typing import List, Union, Tuple
+from argparse import Namespace
+from typing import List, Optional, Union, Tuple
+import warnings
 
 import numpy as np
 from rdkit import Chem
 import torch
 import torch.nn as nn
 
-from .mpn import MPN
 from chemprop.args import TrainArgs
 from chemprop.features import BatchMolGraph
 from chemprop.nn_utils import get_activation_function, initialize_weights
+from chemprop.models.mpn import MPN
 
+class nn_exp(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
 
-class MoleculeModel(nn.Module):
+    def forward(self, x):
+        return torch.exp(x)
+
+class _MoleculeModel(nn.Module):
     """A :class:`MoleculeModel` is a model which contains a message passing network following by feed-forward layers."""
 
-    def __init__(self, args: TrainArgs):
-        """
-        :param args: A :class:`~chemprop.args.TrainArgs` object containing model arguments.
-        """
-        super(MoleculeModel, self).__init__()
+    def __init__(
+        self,
+        dataset_type: str,
+        num_tasks: int,
```

Comment:
`num_tasks` seems to be the source of the errors in testing. It isn't an attribute of args so it's not getting passed through the `**args` in MoleculeModel. It's instead a property function of args, which seems not to get passed.

Several of these others are also property functions but won't necessarily trigger errors because they have default values.

Reply:
ah thanks for that. I was being lazy and just using `**vars(args)` but I'll just spell it all out